sdk/formrecognizer/Azure.AI.FormRecognizer/tests/Models/FormRecognizerModelFactoryTests.cs

```diff
@@ -0,0 +1,527 @@
+            Assert.AreEqual(trainingCompletedOn, customFormModelInfo.TrainingCompletedOn);
+            Assert.AreEqual(status, customFormModelInfo.Status);
+        }
+
+        [Test]
+        public void FormRecognizerModelFactoryCanInstantiateCustomFormSubmodel()
+        {
+            var customFormModelField = new CustomFormModelField("", default);
+
+            var formType = "Pythagoras";
+            var accuracy = 0.1414f;
+            var fields = new Dictionary<string, CustomFormModelField>() { { "", customFormModelField } };
+
+            var customFormSubmodel = FormRecognizerModelFactory.CustomFormSubmodel(formType, accuracy, fields);
+
+            Assert.AreEqual(formType, customFormSubmodel.FormType);
+            Assert.AreEqual(accuracy, customFormSubmodel.Accuracy);
+            Assert.AreNotSame(fields, customFormSubmodel.Fields);
+            Assert.AreEqual(fields, customFormSubmodel.Fields);
+        }
+
+        [Test]
+        public void FormRecognizerModelFactoryCanInstantiateFieldData()
+        {
+            var formElement = new FormWord(default, default, default, default);
+
+            var boundingBox = new BoundingBox(ListOfPoints);
+            var pageNumber = 109;
+            var text = "Poincare";
+            var fieldElements = new List<FormElement>() { formElement };
+
+            var fieldData = FormRecognizerModelFactory.FieldData(boundingBox, pageNumber, text, fieldElements);
+
+            Assert.AreEqual(boundingBox, fieldData.BoundingBox);
+            Assert.AreEqual(pageNumber, fieldData.PageNumber);
+            Assert.AreEqual(text, fieldData.Text);
+            Assert.AreNotSame(fieldElements, fieldData.FieldElements);
+            Assert.AreEqual(fieldElements, fieldData.FieldElements);
+        }
+
+        [Test]
+        public void FormRecognizerModelFactoryCanInstantiateFieldValueWithStringValueType()
+        {
+            var value = "Lovelace";
+
+            var fieldValue = FormRecognizerModelFactory.FieldValueWithStringValueType(value);
+
+            Assert.AreEqual(FieldValueType.String, fieldValue.ValueType);
+            Assert.AreEqual(value, fieldValue.AsString());
+
+            Assert.Throws<InvalidOperationException>(() => fieldValue.AsInt64());
+            Assert.Throws<InvalidOperationException>(() => fieldValue.AsFloat());
+            Assert.Throws<InvalidOperationException>(() => fieldValue.AsDate());
+            Assert.Throws<InvalidOperationException>(() => fieldValue.AsTime());
+            Assert.Throws<InvalidOperationException>(() => fieldValue.AsPhoneNumber());
+            Assert.Throws<InvalidOperationException>(() => fieldValue.AsList());
+            Assert.Throws<InvalidOperationException>(() => fieldValue.AsDictionary());
+        }
+
+        [Test]
+        public void FormRecognizerModelFactoryCanInstantiateFieldValueWithInt64ValueType()
+        {
+            var value = 1709;
```

Comment:
for the FieldValue tests, consider using the type instead of `var` just to make it easier to read